src/vmspawn/vmspawn.c

```diff
@@ -595,6 +598,32 @@ static int start_tpm(sd_bus *bus, const char *scope, const char *our_runtime_dir
         return 0;
 }
 
+static int finalize_root(char **ret) {
+        int r;
+        _cleanup_(dissected_image_unrefp) DissectedImage *image = NULL;
+        _cleanup_free_ char *root = NULL;
+
+        assert(ret);
+
+        r = dissect_image_file_and_warn(arg_image, NULL, NULL, NULL, 0, &image);
+        if (r < 0)
+                return log_error_errno(r, "Failed to dissect image: %m");
```

Comment:
the difference between dissect_image_file_and_warn() and dissect_image_file() is that the former already logs on its own, hence you can drop this line here, it would result in duplicate logging